perses/rjmc/geometry.py

```diff
@@ -268,7 +343,8 @@ def _get_torsions(self, atoms_with_positions, new_atom):
                 continue
             if set(atoms_with_positions).issuperset(set(torsion_partners)):
                 eligible_torsions.append(torsion)
-        return torsions
+        eligible_torsions_with_units = [self._add_torsion_units(torsion) if type(torsion.type.phi_k)!=units.Quantity else torsion for torsion in eligible_torsions]
```

Comment:
This might be simplified if you have `self._add_XXXX_units()` handle the checking about whether units need to be added or not.